veps/sig-compute/image-volume-proposal.md

```diff
@@ -0,0 +1,172 @@
+The new method using imageVolumes will only be used once the feature gate is enabled. By default, the feature 
+gate will be off until it is in Beta stage.
+
+Once the feature gate is graduated, we will ensure the old code is removed from virt-controller and virt-launcher. 
+For virt-handler, we will remove the code for initiating new VMs with container disks but will keep the code needed to 
+support existing VMs. Once we are confident that all virtual machines are using the new method and there are no upgrade 
+risks, we will completely remove the old code for supporting VMs from virt-handler.
+
+## API Examples
+
+As stated in the No Goals section, no API changes are included.
+
+## Alternatives
+
+One alternative to remove the current workaround was explored in [this PR](https://github.com/kubevirt/kubevirt/pull/11845). 
+It suggested using shared namespaces between the launcher containers to avoid bind mounts. However, 
+this approach is more complicated and may compromise the isolation of the compute container where the domain runs, as discussed 
+in [here](https://github.com/kubevirt/kubevirt/pull/11845#issuecomment-2413506306).
+
+Another option would be to introduce a new API field to expose the new implementation. However, this would require adding 
+new fields to both the VM and VMI resources and would mean maintaining support for both the old and new implementations 
+for a long time. This would involve updating the CRDs versions, making the upgrade path more complex to support old VMs\VMIs, 
+and removing the old fields. Manual intervention would be required to migrate VMIs to the new approach, and developers 
+would need to adjust many functional tests to the new API. Additionally, this approach would make users aware of implementation 
+details, complicating the API.
+
+The issue with this approach is similar to the problem with the vm.spec.running field, which we’ve been unable to remove
+due to the breaking change it would cause. This demonstrates why adding new API fields to manage the same behavior is not 
+ideal. 
+
+
+## Scalability
+
+This solution can improve KubeVirt VM scalability by reducing the need for extra containers and privileged operations per VM. 
+It will also lower the resources requests and limits by removing the additional containers and their requirements\limits.
+
+## Update/Rollback Compatibility
+
+To make the upgrade path seamless for users, when the feature is graduated, we will temporary continue to support the old 
+method in virt-handler for older launcher pods. We will then migrate the old launcher pods to the new imageVolume method 
+using live migration. This will update the launcher pod spec to use imageVolumes instead of the old sidecar and init 
+containers, and will also update the image to support the new spec.
+
+## Functional Testing Approach
+
+To ensure the new logic works correctly, we will set up a periodic test lane that runs relevant tests with the feature gate 
```

Comment:
Perhaps we can think of some key things to test? WDYT about:
- Upgrade, live-migration from an old launcher to a new one.
- Using custom kernel boot / initrd.
- Obviously, using container-disks.

Reply:
Done